pkg/forklift-api/webhooks/mutating-webhook/mutating-webhook.go

```diff
@@ -0,0 +1,57 @@
+package mutating_webhook
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
+	"github.com/konveyor/forklift-controller/pkg/lib/logging"
+)
+
+var log = logging.WithName("mutating_webhooks")
+
+type mutator interface {
+	Mutate(*admissionv1.AdmissionReview) *admissionv1.AdmissionResponse
+}
+
+func Serve(resp http.ResponseWriter, req *http.Request, m mutator) {
+	review, err := util.GetAdmissionReview(req)
+	if err != nil {
+		log.Error(err, "mutating Serve error")
+		resp.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	response := admissionv1.AdmissionReview{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: review.APIVersion,
+			Kind:       "AdmissionReview",
+		},
+	}
+	reviewResponse := m.Mutate(review)
+	if reviewResponse != nil {
+		response.Response = reviewResponse
+		response.Response.UID = review.Request.UID
+	}
+	// reset the Object and OldObject, they are not needed in a response.
+	review.Request.Object = runtime.RawExtension{}
+	review.Request.OldObject = runtime.RawExtension{}
+
+	responseBytes, err := json.Marshal(response)
+	if err != nil {
+		resp.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if _, err := resp.Write(responseBytes); err != nil {
+		fmt.Println(err)
+		//log.Log.Reason(err).Errorf("failed to write webhook response")
```

Comment:
used for debugging?

Reply:
yes, missed removing some debuting staff, removed.